src/backend/d3d12/PipelineD3D12.cpp

```diff
@@ -0,0 +1,160 @@
+// Copyright 2017 The NXT Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "PipelineD3D12.h"
+
+#include "D3D12Backend.h"
+#include "ShaderModuleD3D12.h"
+#include "PipelineLayoutD3D12.h"
+
+#include <d3dcompiler.h>
+
+using Microsoft::WRL::ComPtr;
+
+namespace backend {
+namespace d3d12 {
+
+    Pipeline::Pipeline(Device* device, PipelineBuilder* builder) : PipelineBase(builder), device(device) {
+        uint32_t compileFlags = 0;
+#if defined(_DEBUG)
+        // Enable better shader debugging with the graphics debugging tools.
+        compileFlags |= D3DCOMPILE_DEBUG | D3DCOMPILE_SKIP_OPTIMIZATION;
+#endif
+
+        if (IsCompute()) {
+          const auto& module = ToBackend(builder->GetStageInfo(nxt::ShaderStage::Compute).module);
+          const auto& entryPoint = builder->GetStageInfo(nxt::ShaderStage::Compute).entryPoint;
+          const auto& hlslSource = module->GetHLSLSource();
+
+          ComPtr<ID3DBlob> compiledShader;
+          ComPtr<ID3DBlob> errors;
+
+          ASSERT_SUCCESS(D3DCompile(
```

Comment:
To help with debugging, we shouldn't assume that this will work (it isn't clear how good spirv-hlsl is still), and instead we should check for errors and print something if there is a failure.